component.go

```diff
@@ -253,7 +261,8 @@ func RunProc(c interface{}) bool {
 								closeHandler(handlers[chosen].onClose)
 							})
 						}
-						return
+						return // TODO: Debug how to ensure enough shutdowns for inputs
+						// in the pool mode case. for now, pool mode won't work properly
```

Comment:
Please file a separate issue for this describing what exactly won't work properly.